panels/updates/cc-updates-panel.c

```diff
@@ -346,119 +428,80 @@ update_active_network (CcUpdatesPanel *self)
-
-      g_signal_handlers_unblock_by_func (self->automatic_updates_switch,
-                                         on_automatic_updates_switch_changed_cb,
-                                         self);
-    }
 }
 
 static void
-load_tariff (CcUpdatesPanel *self)
+update_active_network (CcUpdatesPanel *self)
 {
-  g_autoptr(MwtTariffLoader) tariff_loader = NULL;
-  g_autoptr(GBytes) bytes = NULL;
-  g_autoptr(GError) error = NULL;
-  g_autofree gchar *tariff_path = NULL;
-  gchar *contents = NULL;
-  gsize size = 0;
+  NMAccessPoint *ap;
+  NMConnection *connection;
+  NMDevice *device;
+  const gchar *icon_name;
 
-  g_debug ("Loading tariff");
+  get_active_connection_and_device (self, &device, &connection, &ap);
 
-  tariff_path = g_build_filename (g_get_user_config_dir (), SYSTEM_TARIFF_FILENAME, NULL);
-  g_file_get_contents (tariff_path, &contents, &size, &error);
-  if (error)
-    {
-      if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
-        g_warning ("Error reading tariff file: %s", error->message);
-      return;
-    }
+  /* Setup the new device */
+  if (self->current_device != device)
+    cleanup_signals (self);
 
-  bytes = g_bytes_new_take (contents, size);
-  tariff_loader = mwt_tariff_loader_new ();
-  mwt_tariff_loader_load_from_bytes (tariff_loader, bytes, &error);
-  if (error)
+  if (g_set_object (&self->current_device, device) && device)
     {
-      g_warning ("Error loading tariff: %s", error->message);
-      return;
+      self->changed_id = g_signal_connect_swapped (device,
+                                                   "state-changed",
+                                                   G_CALLBACK (on_network_changed_cb),
+                                                   self);
     }
 
-  g_signal_handlers_block_by_func (self->tariff_editor, on_tariff_changed_cb, self);
-  g_signal_handlers_block_by_func (self->scheduled_updates_switch, on_scheduled_updates_switch_changed_cb, self);
+  /* Icon */
+  icon_name = get_network_status_icon_name (self, device, ap);
+  gtk_image_set_from_icon_name (GTK_IMAGE (self->network_status_icon), icon_name, 4);
 
-  cc_tariff_editor_load_tariff (self->tariff_editor,
-                                mwt_tariff_loader_get_tariff (tariff_loader),
-                                &error);
+  /* Name */
+  if (ap)
+    {
+      GBytes *ssid = nm_access_point_get_ssid (ap);
+      g_autofree gchar *title = nm_utils_ssid_to_utf8 (g_bytes_get_data (ssid, NULL), g_bytes_get_size (ssid));
 
-  if (error)
-    g_warning ("Error loading tariff: %s", error->message);
+      gtk_label_set_label (GTK_LABEL (self->network_name_label), title);
+    }
   else
-    gtk_switch_set_active (GTK_SWITCH (self->scheduled_updates_switch), TRUE);
+    {
+      if (connection)
+        gtk_label_set_label (GTK_LABEL (self->network_name_label), _("Connected"));
+      else
+        gtk_label_set_label (GTK_LABEL (self->network_name_label), _("No active connection"));
```

Comment:
Nitpick: This could be refactored to only have a single `gtk_label_set_label()` call:
```
gtk_label_set_label (GTK_LABEL (self->network_name_label), connection ? _("Connected") : _("No active connection"));
```